src/torcontrol.cpp

```diff
@@ -531,12 +531,20 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
             SetReachable(NET_ONION, true);
         }
 
+        // Validate the target of the hidden service
```

Comment:
Shouldn't this also be changed, a few lines above:

```cpp
        // Now that we know Tor is running setup the proxy for onion addresses
        // if -onion isn't set to something else.
        if (gArgs.GetArg("-onion", "") == "") {
            CService resolved(LookupNumeric("127.0.0.1", 9050));
            proxyType addrOnion = proxyType(resolved, true);
            SetProxy(NET_ONION, addrOnion);
            SetReachable(NET_ONION, true);
        }
```

If the tor daemon is running on another machine, then it will not be reachable on 127.0.0.1:9050.